spec/features/admin/dashboard_spec.rb

```diff
@@ -0,0 +1,95 @@
+
+    @order2 = @user1.orders.create!(name: @user1.name, address: @user1.address, city: @user1.city, state: @user1.state, zip: @user1.zip, status: "Cancelled")
+    @order2.item_orders.create!([{ item: @pull_toy, quantity: 2, price: @pull_toy.price }, { item: @dog_bone, quantity: 3, price: @dog_bone.price }, { item: @spring, quantity: 1, price: @spring.price }])
+
+    @order3 = @user2.orders.create!(name: @user2.name, address: @user2.address, city: @user2.city, state: @user2.state, zip: @user2.zip, status: "Packaged")
+    @order3.item_orders.create!(item: @tire, quantity: 2, price: @tire.price)
+
+    @order4 = @user2.orders.create!(name: @user2.name, address: @user2.address, city: @user2.city, state: @user2.state, zip: @user2.zip, status: "Pending")
+    @order4.item_orders.create!(item: @spring, quantity: 1, price: @spring.price)
+
+    visit '/login'
+
+    fill_in :email, with: @admin.email
+    fill_in :password, with: @admin.password
+
+    click_button 'Login'
+
+    visit '/admin'
+  end
+  describe "when I visit my admin dashboard ('/admin')" do
+    it "I see all orders in the system and each order's information" do
+      within '#all-orders' do
+        within "#order-#{@order1.id}" do
+          expect(page).to have_link(@order1.name)
+          expect(page).to have_content(@order1.id)
+          expect(page).to have_content(@order1.created_at.to_date)
+          expect(page).to have_content(@order1.status)
+        end
+        within "#order-#{@order2.id}" do
+          expect(page).to have_link(@order2.name)
+          expect(page).to have_content(@order2.id)
+          expect(page).to have_content(@order2.created_at.to_date)
+          expect(page).to have_content(@order2.status)
+        end
+        within "#order-#{@order3.id}" do
+          expect(page).to have_link(@order3.name)
+          expect(page).to have_content(@order3.id)
+          expect(page).to have_content(@order3.created_at.to_date)
+          expect(page).to have_content(@order3.status)
+        end
+      end
+    end
+    it "each user name under orders is a link to the admin view of that user's profile" do
+      within "#order-#{@order1.id}" do
+        click_link(@order1.name)
+      end
+
+      expect(current_path).to eq("/admin/users/#{@order1.user_id}")
+
+      visit '/admin'
+
+      within "#order-#{@order2.id}" do
+        click_link(@order2.name)
+      end
+
+      expect(current_path).to eq("/admin/users/#{@order2.user_id}")
+
+      visit '/admin'
+
+      within "#order-#{@order3.id}" do
+        click_link(@order3.name)
+      end
+
+      expect(current_path).to eq("/admin/users/#{@order3.user_id}")
+    end
+    it "orders are sorted by status: packaged, pending, shipped, then cancelled" do
+      within '#all-orders' do
+        # Proper order is: @order3, @order4, @order1, @order2
```

Comment:
Why have I never thought to use the comments feature to remind myself of certain things?